src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -715,15 +715,119 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
                     frontend_logger.log("Error writing to socket", "error");
                 }
             } else {
+                int result_wr = write(connFd, PRIORITY.c_str(), PRIORITY.length());
+                if (result_wr < 0) {
+                    frontend_logger.log("Error writing to socket", "error");
+                }
+                result_wr = write(connFd, "\r\n", 2);
+                if (result_wr < 0) {
+                    frontend_logger.log("Error writing to socket", "error");
+                }
+
+                // We get the name and the path to graph as a pair separated by |.
+                char priority_data[300];
+                bzero(priority_data, 301);
+
+                read(connFd, priority_data, FRONTEND_DATA_LENGTH);
+
+                string priority(priority_data);
+
+                Utils utils;
+                priority = utils.trim_copy(priority, " \f\n\r\t\v");
+
+                if (!(std::find_if(priority.begin(),
+                                        priority.end(), [](unsigned char c) { return !std::isdigit(c); }) == priority.end())) {
+                    string error_message = "Priority should be numeric and > 1 or empty";
+                    result_wr = write(connFd, error_message.c_str(), error_message.length());
+
+                    if (result_wr < 0) {
+                        frontend_logger.log("Error writing to socket", "error");
+                    }
+
+                    result_wr = write(connFd, "\r\n", 2);
+
+                    if (result_wr < 0) {
+                        frontend_logger.log("Error writing to socket", "error");
+                    }
+                    break;
+                }
+
+                int threadPriority = std::atoi(priority.c_str());
+
+                //All high priority threads will be set the same high priority level
+                if (threadPriority > Conts::DEFAULT_THREAD_PRIORITY) {
+                    threadPriority = Conts::HIGH_PRIORITY_DEFAULT_VALUE;
+
+                    if (highPriorityTaskCount > 0) {
+                        bool rejectJob = false;
+                        PerformanceUtil performanceUtil;
+                        performanceUtil.init();
+
+                        bool resourceSufficient = performanceUtil.isResourcesSufficient(graph_id,TRIANGLES,Conts::SLA_CATEGORY::LATENCY);
+                        int runningHPTaskCount = JasmineGraphFrontEnd::runningHighPriorityTasks();
```

Comment:
Can we change the name of this function `runningHighPriorityTasks()` to `getRunningHighPriorityTasksCount()`. I think the new name is more meaningful.

Reply:
I've changed the method name